bundles/core/org.openhab.core/src/main/java/org/openhab/core/types/UnDefType.java

```diff
@@ -34,4 +34,20 @@ public String format(String pattern) {
 		return String.format(pattern, this.toString());
 	}
 
+	/**
+	 * Returns the enum constant with the specified name.
+	 */
+	public static UnDefType parse(String name) {
+		if (name == null) {
+			throw new NullPointerException("Name is null");
+		}
+		if (UNDEF.toString().equals(name)) {
+			return UNDEF;
+		} else if (NULL.toString().equals(name)) {
+			return NULL;
+		}
+		throw new IllegalArgumentException("No enum constant "
+				+ UnDefType.class.getCanonicalName() + "." + name);
+	}

```

Comment:
> I found this problem while i wrote a jsr223 script, i wanted to update a item:
>   ```postUpdate("ITEM_NAME", UnDefType.UNDEF);```
> i got the IllegalArgumentException and searched the reason

@gerrieg In JSR223, the postUpdate overloads are defined on the BusEvent class. The only signature taking a string as the first argument also requires a string for the second argument. 

```postUpdate(String, String)```

Could that be why you are seeing the IllegalArgumentException (although I saw a different error when I tried the incorrect code)? Note that you can't just do ```UnDefType.UNDEF.toString()``` for the second argument because depending on the item type it might match another acceptable type. StringItem is one example where that would happen.

When I do ```postUpdate(ITEM, UnDefType.UNDEF)``` (item instance instead of name), it works as expected.